### ISSUE TITLE
[Question] Failing tests

### ISSUE THREAD

**Issue by ehhc** [AbdulRahmanAlHamali/flutter_pagewise#34]
Hey guys, 
Thanks for this cool library. I'm using it and i want to write WidgetTests for my app. The following shows a minimal working example of the app. Unfortunately my test is failing and i don't understand why. Can you help me by explaining why? How do i get my test running?

My App:
```
import 'package:flutter/material.dart';
import 'package:flutter_pagewise/flutter_pagewise.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: PageWiseWidget()
    );
  }
}

class PageWiseWidget extends StatefulWidget{
  @override
  PageWiseWidgetState createState() {
    return PageWiseWidgetState();
  }
}

class PageWiseWidgetState extends State<PageWiseWidget> {
  @override
  Widget build(BuildContext context) {
    return
      Column(
          children: [
            Expanded(
                child: PagewiseListView(
                  pageFuture: BackendService.getPage,
                  pageSize: 10,
                  itemBuilder: (context, entry, _) => Text(entry['name']),
                  noItemsFoundBuilder: (context) => Text('Keine passenden Kliniken gefunden'),
                )
            ),
          ]
      );
  }
}

class BackendService {
  static Future<List> getPage(pageIndex) async {
    List list = List.generate(pageIndex < 6 ? 10 : 5, (index) {
      int dataNumber = index + pageIndex * 10;
      return {
        'name': 'product' + dataNumber.toString(),
      };
    });
    return list;
  }
}
```

The failing test:
```

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:page_wise_test/main.dart';

void main() {
  testWidgets('Elements are shown', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(StatefulBuilder(
        builder: (BuildContext context, StateSetter setState) {
          return MaterialApp(
            home: Material(
                child: PageWiseWidget()
            ),
          );
        })
    );
    expect(find.text('product3'), findsOneWidget);
  });
}
```

The error message is:
`══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure object was thrown running a test:
  Expected: exactly one matching node in the widget tree
  Actual: ?:<zero widgets with text "product3" (ignoring offstage widgets)>
   Which: means none were found but one was expected
`

Do you have any idea why the product is shown in the running app but is not visible during the tests?
Thanks a lot :)

**Comment by AbdulRahmanAlHamali** [comment]
Hmmmm I am not sure. The Pagewise components only load their new components when we scroll down to them. This test could be failing because the "scrolling" is not happening, but I cannot confirm that.

I will try to test it in a couple of days and will get back to you

**Comment by AbdulRahmanAlHamali** [comment]
Hello @ehhc, I apologize but I have not been able to test your issue yet. Were you able to make any progress?

**Comment by ehhc** [comment]
hi @AbdulRahmanAlHamali thanks for your answer. No i wasn't able to make any progress.. The test (as given in the first post) fails and i'm not sure how to work around that.. Nonetheless, i noticed that the problem is not as urgent as i initially thought. I noticed that my main problem was not directly caused by your library but because i'm new to flutter and misunderstood something. Anyway: i think in the long run it would be nice to have some tests for your library as well. Furtunately it aint that pressing anymore :)

**Comment by AbdulRahmanAlHamali** [comment]
Alright great to know you're not blocked by anything!

And I definitely agree, this is on the roadmap, and will let you know once your issue is solved!

**Comment by ehhc** [comment]
Thanks again :)
